### PR TITLE
Allow multiple instances registering

### PR DESCRIPTION
**Pull request by darkgl0w**
Address and close https://github.com/fastify/fastify-redis/issues/28#issue-439226579

No breaking changes were introduced with this PR and the external client behavior has been extended to the named instances.

#### Checklist

- [x] run `npm run test`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by darkgl0w** [comment]
Hi @Eomm ^^
I implemented one more check to prevent the behavior you described (as it may be considered a bad practice :d) and I made the plugin throw an error. I also updated the documentation to point out the fact that this new feature implies to namespace all the registered instances of `fastify-redis`.

Also it's to be noted that using this feature won't be a breaking change (at least not really a breaking change) but it requires a little bit of refactoring on existing projects if someone wants to use it. ^^

File: test.js [hunk 1]
```diff
@@ -110,3 +153,96 @@ test('custom client', t => {
...
+    })
+    .register(fastifyRedis, {
+      host: '127.0.0.1',
+      namespace: 'test'
+    })
+
+  fastify.ready((err) => {
+    t.is(err.message, 'A Redis instance has already been registered without a namespace')
+  })
+})
```
Comment (Eomm) [inline test.js]: Sorry, I didn't explain well my point, this is what I intended for 👍
So we can test that a user can start using this new feature without breaking some others plugins

```suggestion
})

test('Should not throw in with different contexts', (t) => {
  t.plan(1)

  const fastify = Fastify()
  t.teardown(() => fastify.close())

  fastify.register(function (instance, options, next) {
    fastify.register(fastifyRedis, {
      host: '127.0.0.1'
    })
    next()
  })

  fastify.register(function (instance, options, next) {
    fastify
      .register(fastifyRedis, {
        host: '127.0.0.1',
        namespace: 'test1'
      })
      .register(fastifyRedis, {
        host: '127.0.0.1',
        namespace: 'test2'
      })
    next()
  })

  fastify.ready(t.error)
})
```

Comment (darkgl0w) [reply test.js]: Oh ok I see now :rofl:

**Comment by darkgl0w** [comment]
Applied all the necessary changes and corrected the faulty test.

**Pull request merged**